File: Longest Substring Without Repeating Characters.java
import java.util.* ;
import java.io.*; 
public class Solution {
        
	public static int lengthOfLongestSubstring(String s) {


		int n=s.length();
		Map<Character,Integer>mp=new HashMap<>();
     int mx=0,c=0;
	 int j=0;
		for(int i=0;i<n;i++)
		{

			int val=0;
			c++;
			char ch=s.charAt(i);
			if(mp.containsKey(ch))
			{
				val=mp.get(ch);
			}
			mp.put(s.charAt(i),val+1);
			if(val+1==2)
			{
				val=2;
				while(val+1>2)
				{
					ch=s.charAt(j);
					int d=mp.get(ch);
					mp.put(ch,d-1);
					ch=s.charAt(i);
					val=mp.get(ch);
					j++;
					c--;
				}
			}
			mx=Math.max(mx,c);
		}
		return mx;
		// Write your code here.
	}
}

File: Longest Substring Without Repeating Characters.cpp
#include <bits/stdc++.h>
int lengthOfLongestSubstring(string &s) {

map<char,int>mp;
    int mx=0,c=0;
    int j=0;
    for(int i=0;i<s.size();i++)
    {
        mp[s[i]]++;
        c++;
        if(mp[s[i]]==2){
          while(mp[s[i]]==2)
          {
              mp[s[j]]--;
              c--;
              j++;
          }

        }
        mx=max(mx,c);

    }
    return mx;
    // Write your code here.
}
